src/HIE/Bios/Cradle.hs

```diff
@@ -128,9 +133,22 @@ implicitConfig' fp = (\wdir ->
          (Bios (wdir </> ".hie-bios") Nothing, wdir)) <$> biosWorkDir fp
   --   <|> (Obelisk,) <$> obeliskWorkDir fp
   --   <|> (Bazel,) <$> rulesHaskellWorkDir fp
-     <|> (stackExecutable >> (Stack Nothing,) <$> stackWorkDir fp)
-     <|> ((Cabal Nothing,) <$> cabalWorkDir fp)
-
+     <|> (cabalExecutable >> cabalProjectDir fp >> cabalDistDir fp >>= cabal)
```

Comment:
There is a major bug when I run this on the haskell-language-server repo hie-bios picks cabal before getting here. I tried changing the order so stack is first, but it picks before `implicitConfig'`

```sh
❯ rm -rf dist-newstyle
❯ stack build
❯ hie-bios debug src/Ide/Cradle.hs         
Cradle failed to load
Exit Code: ExitFailure 1
Stderr: Failed to parse result of calling cabal
Resolving dependencies...

From https://github.com/Avi-D-coder/hie-bios
 + 6d95e03...63e6e56 master     -> origin/master  (forced update)
Previous HEAD position was b67ebfd Use single file hie-implicit
HEAD is now at 63e6e56 Use single file hie-implicit
Warning: Requested index-state2020-05-05T17:33:00Z is newer than
'hackage.haskell.org'! Falling back to older state (2020-05-02T22:21:20Z).
cabal: Could not resolve dependencies:
[__0] trying: ghcide-0.1.0 (user goal)
[__1] next goal: haskell-lsp-types (dependency of ghcide)
[__1] rejecting: haskell-lsp-types-0.21.0.0, haskell-lsp-types-0.20.0.0,
haskell-lsp-types-0.19.0.0, haskell-lsp-types-0.18.0.0,
haskell-lsp-types-0.17.0.0, haskell-lsp-types-0.16.0.0,
haskell-lsp-types-0.15.0.0, haskell-lsp-types-0.14.0.1,
haskell-lsp-types-0.14.0.0, haskell-lsp-types-0.13.0.0,
haskell-lsp-types-0.12.1.0, haskell-lsp-types-0.12.0.0,
haskell-lsp-types-0.11.0.0, haskell-lsp-types-0.10.0.0,
haskell-lsp-types-0.9.0.0, haskell-lsp-types-0.8.2.0,
haskell-lsp-types-0.8.0.1, haskell-lsp-types-0.8.0.0,
haskell-lsp-types-0.7.0.0, haskell-lsp-types-0.6.0.0,
haskell-lsp-types-0.5.0.0, haskell-lsp-types-0.4.0.0,
haskell-lsp-types-0.2.2.0, haskell-lsp-types-0.2.1.0,
haskell-lsp-types-0.15.0.1 (conflict: ghcide => haskell-lsp-types==0.22.*)
[__1] fail (backjumping, conflict set: ghcide, haskell-lsp-types)
After searching the rest of the dependency tree exhaustively, these were the
goals I've had most trouble fulfilling: ghcide, haskell-lsp-types
```